src/main/rcsplit/rcsplit.c

```diff
@@ -0,0 +1,291 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <ctype.h>
+
+#include <platform.h>
+
+#include "common/utils.h"
+
+#include "config/parameter_group.h"
+#include "config/parameter_group_ids.h"
+
+#include "fc/rc_controls.h"
+
+#include "io/beeper.h"
+#include "io/serial.h"
+
+#include "drivers/serial.h"
+
+#include "rcsplit/rcsplit.h"
+
+
+#define STATIC_ASSERT(condition, name) \
+    typedef char assert_failed_ ## name [(condition) ? 1 : -1 ]
+
+// This is public setting in the RunCam Split tab of CleanFlight Configurator
+// The user can bind the following mode with some aux channel in special range.
+static const box_t rcsplitBoxes[RCSPLIT_CHECKBOX_ITEM_COUNT] = {
+    { RCSPLIT_BOX_SIM_WIFI_BUTTON, "Wi-Fi Button", 0 },
+    { RCSPLIT_BOX_SIM_POWER_BUTTON, "Power Button", 1 },
+    { RCSPLIT_BOX_SIM_CHANGE_MODE, "Change Mode", 2 },
+};
+
+// mask of enabled RunCam Split Box IDs, calculated on startup based on enabled features. rcsplitBoxId_e is used as bit index
+uint32_t rcsplitModeActivationMask; // one bit per mode defined in rcsplitBoxId_e
+
+
+static uint32_t rcsplitActiveBoxIds;
+STATIC_ASSERT(sizeof(rcsplitActiveBoxIds) * 8 >= RCSPLIT_CHECKBOX_ITEM_COUNT, RCSPLIT_CHECKBOX_ITEMS_wont_fit_in_activeBoxIds);
+
+// the rcsplit mode setting in the EEPRom
+PG_REGISTER_ARRAY(modeActivationCondition_t, MAX_RC_SPLIT_MODE_ACTIVATION_CONDITION_COUNT, rcsplitModeActivationConditions, PG_RCSPLIT_MODE_ACTIVATION_PROFILE, 0);
+
+// communicate with camera device variables
+static serialPort_t *rcsplitSerialPort = NULL;
+static rcsplit_switch_state_t switchStates[RCSPLIT_CHECKBOX_ITEM_COUNT];
+static rcsplit_state_e cameraState = RCSPLIT_STATE_UNKNOWN;
+
+static unsigned char crc_high_first(unsigned char *ptr, unsigned char len)
+{
+    unsigned char i; 
+    unsigned char crc=0x00;
+    while(len--) {
+        crc ^= *ptr++;  
+        for (i=8; i>0; --i) { 
+            if (crc & 0x80)
+                crc = (crc << 1) ^ 0x31;
+            else
+                crc = (crc << 1);
+        }
+    }
+    return (crc); 
+}
+
+static void sendCtrlCommand(rcsplit_ctrl_cmd_argument_e argument)
+{
+    if (!rcsplitSerialPort)
+        return ;
+
+    uint8_t tmpBuffer[4];
+    unsigned char uart_buffer[5] = {0};
+    unsigned char crc = 0;
+
+    // first calc crc with the request, 
+    // note: the componnents that need to crc is [header]+[command]+[argument]+[tail], 
+    // no need calc  with crc field, event the crc field is zero
+    tmpBuffer[0] = RCSPLIT_PACKET_HEADER;
+    tmpBuffer[1] = RCSPLIT_PACKET_CMD_CTRL;
+    tmpBuffer[2] = argument;
+    tmpBuffer[3] = RCSPLIT_PACKET_TAIL;
+
+    crc = crc_high_first(tmpBuffer, 4);
+
+    // build up a full request [header]+[command]+[argument]+[crc]+[tail]
```

Comment:
ah..that is not a good idea to hard code the crc. 

Because it wants to reduce the size of the code, otherwise the implementation of .travis.sh script will always fail. You are right, I will restore the crc calculation